board/tests.py

```diff
@@ -214,11 +216,197 @@ def test_get_posts(self):
+                'board': str(b.uuid),
+                'name': 'bob',
+                'message': 'hello world',
+                'photo': self.get_test_image('test.png'),
+            },
+            HTTP_ACCEPT='application/json',
+        )
+
+        p = b.post_set.all()[0]
+        # Ensuring it is properly saved in the database.
+        self.assertEqual(p.name, 'bob')
+        self.assertEqual(p.message, 'hello world')
+        self.assertEqual(p.photo.name, 'test.png')
+        self.assertEqual(p.photo.photo, self.get_test_image('test.png').read())
+
+        exp = {
+            'name': 'bob',
+            'message': 'hello world',
+            'photo': {
+                'uuid': str(p.photo.uuid),
+                'name': 'test.png',
+            }
+        }
+        self.assertEqual(res.json(), exp)
+
+
+    @tag('core')
+    def test_create_posts_optional_fields(self):
+        """Creating a post with only certain optional fields given."""
+
+        b = Board(title='hi', description='hello')
+        b.save()
+
+        # Name + message field.
+        res1 = self.client.post(
+            reverse('board:post-create'),
+            {'board': str(b.uuid), 'name': 'bob', 'message': 'hello world'},
+            HTTP_ACCEPT='application/json',
+        )
+        exp1 = {
+            'name': 'bob', 
+            'message': 'hello world',
+            'photo': None,
+        }
+        
+        # Message field only.
+        time.sleep(0.01) # Ensuring that there is a proper delay for the requests.
+        res2 = self.client.post(
+            reverse('board:post-create'),
+            {'board': str(b.uuid), 'message': 'hello world2'},
+            HTTP_ACCEPT='application/json',
+        )
+        exp2 = {
+            'name': '', 
+            'message': 'hello world2',
+            'photo': None,
+        }
+        # Photo field only.
+        time.sleep(0.01) # Ensuring that there is a proper delay for the requests.
+        res3 = self.client.post(
+            reverse('board:post-create'),
+            {'board': str(b.uuid), 'photo': self.get_test_image('test.png')},
+            HTTP_ACCEPT='application/json',
+        )
+        
+
+        self.assertEqual(res1.json(), exp1)
+        self.assertEqual(res2.json(), exp2)
+          
+
+        # Ensure that they're saved to the database properly.
+        posts = b.post_set.all().order_by('created_at')
+        p1, p2, p3 = posts
+
+        self.assertEqual(p1.name, 'bob')
+        self.assertEqual(p1.message, 'hello world')
+        self.assertIs(p1.photo, None)
+
+        self.assertIs(p2.name, '')
+        self.assertEqual(p2.message, 'hello world2')
+        self.assertIs(p2.photo, None)
+
+        self.assertIs(p3.name, '')
+        self.assertIs(p3.message, '')
+        self.assertEqual(p3.photo.name, 'test.png')
+        self.assertEqual(p3.photo.photo, self.get_test_image('test.png').read())
+        exp3 = {
+            'name': '', 
+            'message': '',
```

Comment:
is message being `''` enforced on the API level (that is, if I give you null or None for message, it gets swapped to `''`), or do you assume that the forms are giving you `''` for no message?

If so make sure to document it somewhere if you haven't already

Reply:
`exp3` is the expected output, look at line 296 for what the client gives to the server.

Reply:
Is it documented somewhere? I haven't looked too much at the top code or whatever but as long as it is clearly outlined that `''` is expected it should be okay

Reply:
I made it into a documentation issue